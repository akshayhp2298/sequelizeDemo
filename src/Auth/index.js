import jwt from "jsonwebtoken"

exports.getToken = user => {
  return jwt.sign(user, process.env.SECRETKEY)
}
export default jwt