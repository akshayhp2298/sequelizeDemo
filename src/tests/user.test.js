import request from "./request"
import userQuery from "../../DB/query/user"
import * as auth from "../Auth/index"

const user = {
  email: "sample@123.com",
  password: "root",
  firstName: "test",
  lastName: "test"
}
const jwt = auth.getToken(user, process.env.SECRETKEY)
describe("user route test", () => {
  userQuery.getUserByEmail = jest.fn(() => ({
    id: 3,
    firstName: "test",
    lastName: "test",
    email: user.email,
    password: "63a9f0ea7bb98050796b649e85481845"
  }))
  it("should return done", async () => {
    const res = await request.get("/api")
    expect(res.body.done).toBe(true)
  })

  it("should create user", async () => {
    userQuery.createUser = jest.fn()
    const res = await request.post("/api/user").send(user)
    expect(res.status).toBe(200)
  })

  it("should update user", async () => {
    const res = await request
      .put("/api/user")
      .set("Authorization", jwt)
      .send({ ...user, firstName: "changed" })
    expect(res.status).toBe(200)
    expect(res.body).toMatchSnapshot()
  })

  it("should return user", async () => {
    const res = await request
      .get("/api/user")
      .set("Authorization", jwt)
      .send(user)
    expect(res.status).toBe(200)
    expect(res.body).toMatchSnapshot()
  })

  it("should return unauthorized", async () => {
    const res = await request
      .get("/api/user")
      .set("Authorization", `${jwt}asd`)
      .send(user)
    expect(res.status).toBe(401)
  })

  it("should login user and return token", async () => {
    const res = await request
      .post("/api/user/login")
      .send({ email: user.email, password: user.password })
    expect(res.status).toBe(200)
    expect(res.body).toHaveProperty("token")
  })

  it("should return password not valid for login", async () => {
    const res = await request
      .post("/api/user/login")
      .send({ email: user.email, password: 'asd' })
    expect(res.status).toBe(401)
  })

  it("should update user", async () => {
    userQuery.updateUser = jest.fn()
    const res = await request
      .put("/api/user")
      .set("Authorization", jwt)
      .send({ ...user, firstName: "changed" })
    expect(res.status).toBe(200)
    expect(res.body).toMatchSnapshot()
  })
})
