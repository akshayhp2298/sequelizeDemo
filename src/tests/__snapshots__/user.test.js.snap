// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user route test should return user 1`] = `
Object {
  "message": "user data",
  "user": Object {
    "email": "sample@123.com",
    "firstName": "test",
    "id": 3,
    "lastName": "test",
  },
}
`;

exports[`user route test should update user 1`] = `
Object {
  "message": "user updated",
}
`;

exports[`user route test should update user 2`] = `
Object {
  "message": "user updated",
}
`;
